File: Medium/2419. Longest Subarray With Maximum Bitwise AND.java
class Solution {
    public int longestSubarray(int[] nums) {
        int max = 0, res = 0, len = 0;
        for (int num : nums) {
            max = Math.max(max, num);
        }
        for (int num : nums) {
            res = Math.max(res, len = num < max ? 0 : len + 1);
        }
        return res;
    }
}

File: Medium/2419. Longest Subarray With Maximum Bitwise AND.py
from typing import List


class Solution:
    def longestSubarray(self, nums: List[int]) -> int:
        max_num = streak = longest_subarray = 0
        for num in nums:
            if num == max_num:
                streak += 1
                longest_subarray = max(longest_subarray, streak)
            elif num > max_num:
                max_num = num
                longest_subarray = streak = 1
            else:
                streak = 0
        return longest_subarray